main-settings/src/test/scala/sbt/TupleSyntaxTest.scala

```diff
@@ -0,0 +1,18 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt.test
+
+import sbt._
+import sbt.Def.Initialize
+import sbt.TupleSyntax._
+
+object TupleSyntaxTest {
+  def t1[T](a: SettingKey[T], b: TaskKey[T], c: Initialize[T], d: Initialize[Task[T]]) = {
+    (a, b, c.toTaskable, d.toTaskable).map((x: T, y: T, z: T, w: T) => "" + x + y + z + w)
```

Comment:
Could we somehow remove `toTaskable` method, and put the implicit conversion under `import sbt.TupleSyntax._`?

Reply:
No, I tried, hard, with a bunch of abstract types and so forth (because `Task` doesn't exist until `task-sytem` etc).  The problem is that inheritance makes it impossible.

`SettingKey[T] <: Initialize[T]`
`SettingKey[T] <: Taskable[T]`
`TaskKey[T] <: Initialize[Task[T]]`
`TaskKey[T] <: Taskable[T]`

So you can't define a class hierarchy and use that as the `ST` in `TupleSyntax` that works for SettingKey, TaskKey, Initialize[Task[T]] and any old Initialize[T].  And implicit conversions don't chain automatically.  And obviously we don't want to do all the combinations of those 4 in tuples2-tuple11.

The (prioritised) implicit conversions in `Taskable`'s implicit scope will trigger for things like `Classpaths.concat`, because `Taskable` is the target type, not a prerequisite for the tuple enrichments.

Reply:
ok.